packages/test/test-end-to-end-tests/src/test/gc/gcUnreferencePhases.spec.ts

```diff
@@ -0,0 +1,133 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import {
+    createSummarizerWithContainer,
+    ITestContainerConfig,
+    ITestObjectProvider,
+    mockConfigProvider,
+    summarizeNow,
+    waitForContainerConnection,
+} from "@fluidframework/test-utils";
+import { describeNoCompat, ITestDataObject, TestDataObjectType } from "@fluidframework/test-version-utils";
+import { IGCRuntimeOptions } from "@fluidframework/container-runtime";
+import { delay, stringToBuffer } from "@fluidframework/common-utils";
+import { gcTreeKey } from "@fluidframework/runtime-definitions";
+import { SummaryType } from "@fluidframework/protocol-definitions";
+import { getGCStateFromSummary, getGCTombstoneStateFromSummary } from "./gcTestSummaryUtils";
+
+/**
+ * Validates that the unreferenced timestamp is correctly set in the GC summary tree. Also, the timestamp is removed
+ * when an unreferenced node becomes referenced again.
+ */
+describeNoCompat("GC unreference phases", (getTestObjectProvider) => {
+    const inactiveTimeoutMs = 100;
+    const sweepTimeoutMs = 200;
+
+    const settings = {};
+    const gcOptions: IGCRuntimeOptions = {
+        inactiveTimeoutMs,
+        sweepTimeoutMs,
+    };
+    const testContainerConfig: ITestContainerConfig = {
+        runtimeOptions: {
+            summaryOptions: {
+                summaryConfigOverrides: {
+                    state: "disabled",
+                },
+            },
+            gcOptions,
+        },
+        loaderProps: { configProvider: mockConfigProvider(settings) },
+    };
+
+    let provider: ITestObjectProvider;
+    let documentAbsoluteUrl: string | undefined;
+
+    const loadSummarizerAndContainer = async (summaryVersion?: string) => {
+        return createSummarizerWithContainer(
+            provider,
+            documentAbsoluteUrl,
+            summaryVersion,
+            gcOptions,
+            mockConfigProvider(settings),
+        );
+    };
+
+    beforeEach(async function() {
+        provider = getTestObjectProvider({ syncSummarizer: true });
+        // These tests validate the GC state in summary generated by the container runtime. They do not care
+        // about the snapshot that is downloaded from the server. So, it doesn't need to run against real services.
+        if (provider.driver.type !== "local") {
+            this.skip();
+        }
+
+        settings["Fluid.GarbageCollection.ThrowOnTombstoneUsage"] = true;
+        settings["Fluid.GarbageCollection.TestOverride.SweepTimeoutMs"] = sweepTimeoutMs;
+    });
+
+    it("GC nodes go from referenced to unreferenced to inactive to sweep ready to tombstone", async () => {
```

Comment:
I think moving the test to `gcTombstoneAttachmentBlobs.spec.ts` and `gcTombstoneDataStores.spec.ts`  for blob and data store respectively would be better since all tombstone tests will be one place.

Reply:
I think I was planning on adding sweep to this test eventually, so this test may become a sweep test.

Reply:
I would expect the test would be needed for both tombstone and sweep.